rclc/CMakeLists.txt

```diff
@@ -1,56 +1,126 @@
+find_package(std_msgs REQUIRED)
 find_package(rosidl_generator_c REQUIRED)
 
-ament_export_include_directories(include)
-include_directories(include)
 
-if(NOT CMAKE_C_STANDARD)
-  set(CMAKE_C_STANDARD 11)
-endif()
+#################################################
+# include directories
+#################################################
+include_directories(include)
 
-set(rclc_sources
-  "src/functions.c"
-)
+#################################################
+# create library
+#################################################
+add_library(${PROJECT_NAME} src/init.c src/publisher.c src/subscription.c src/timer.c src/node.c src/executor_handle.c src/executor.c)
 
-add_library(${PROJECT_NAME}
-  SHARED
-  ${rclc_sources}
+target_include_directories(${PROJECT_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
 )
 
 ament_target_dependencies(${PROJECT_NAME}
   rcl
+  rcutils
   rosidl_generator_c
 )
 
-ament_export_dependencies(
-  rcl
-  rmw
-  rosidl_generator_c
-)
+
+#################################################
+# install
+#################################################
+
+install(DIRECTORY include/ DESTINATION include)
+install(TARGETS ${PROJECT_NAME} DESTINATION lib)
+
+# specific order: dependents before dependencies
 ament_export_include_directories(include)
 ament_export_libraries(${PROJECT_NAME})
 
+#################################################
+# gtest
+#################################################
+
 if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
   find_package(ament_lint_auto REQUIRED)
+  set(ament_cmake_copyright_FOUND TRUE) # No copyright header check since link to NOTICE file is not recognized properly.
+  find_package(std_msgs REQUIRED)
+  find_package(geometry_msgs REQUIRED)
+  find_package(osrf_testing_tools_cpp REQUIRED)
   ament_lint_auto_find_test_dependencies()
 
-  add_subdirectory(test)
+  ament_add_gtest(${PROJECT_NAME}_test
+    test/test_init.cpp
+    test/test_node.cpp
+    test/test_publisher.cpp
+    test/test_subscription.cpp
+    test/test_timer.cpp
+    test/test_executor_handle.cpp
+    test/test_executor.cpp
+  )
+
+  target_include_directories(${PROJECT_NAME}_test PRIVATE include)
+  target_link_libraries(${PROJECT_NAME}_test ${PROJECT_NAME})
+  ament_target_dependencies(${PROJECT_NAME}_test
+    rcl
+    rcutils
+    rosidl_generator_c
+    osrf_testing_tools_cpp
+    std_msgs
+    geometry_msgs
+  )
 endif()
 
+#################################################
+# export dependencies
+#################################################
+# specific order: dependents before dependencies
+ament_export_include_directories(include)
+ament_export_libraries(${PROJECT_NAME})
+ament_export_dependencies(ament_cmake)
+ament_export_dependencies(rcl)
+ament_export_dependencies(rcutils)
+ament_export_dependencies(rosidl_generator_c)
 ament_package()
 
 install(
   DIRECTORY include/
   DESTINATION include
 )
```

Comment:
Isn't this redundant with line 69?

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b